### ISSUE TITLE
AWI problem: no URL when clicking post age to display post

### ISSUE THREAD

**Issue by mikehamca** [mastodon/mastodon#27137]
### Steps to reproduce the problem

1. Start with a frresh tab. Open Mastodon with the Advanced Web Interface.
2. Click the "age" of a post — e.g., "3 min" — to display URL.
3. URL displays — example: "https://mstdn.ca/deck/@ned/111127230230765355"
4. Remove "deck/" if you want to copy the URL (e.g., for a message) — that's not the problem I'm reporting, but I sue wish "deck/" was not inserted into the URL.
5. Delete the URL. 
6. Go to another post. Click the age. 
7. Post is displayed in the righthand column, but URL field is blank.(I believe the URL should now be the URL of the post just displayed — and without "deck/")
8. Reload screen.
9. URLs now resume reappearing when age is clicked (though always with "deck/" inserted).
10. The problem is that reloading the screen brings in new messages and loses one's place in the timeline.
...

### Expected behaviour

Deleting the URL for one post does not affect URL display for another post.

### Actual behaviour

Deleting the URaL for one post prevents URL display for subsequent posts.

### Detailed description

If I delete the URL for a post (after clicking its age), I cannot get the URL for any post to display unless I reload the page, which brings in new posts and loses my post.

I think that whenever I click the post age (in the Advanced Web Interface), then post should display and the URL of the post should appear in the address bar, and that URL should not have "deck/l" included. 

### Mastodon instance

mstdn.ca

### Mastodon version

v4.2.0+glitch

### Browser name and version

Vivaldi	6.2.3105.51 (Stable channel) (arm64)  Revision	76144063ae8137b05a189fe91dfaf1ac12ce492d

### Operating system

macOS Version 13.6 (Build 22G120)

**Comment by ClearlyClaire** [comment]
If I understand correctly, your issue is that once you've started manually editing the URL in your browser, it doesn't update on navigation? In that case, this is your browser's behavior, and I do not think we can do anything about it.

**Comment by ClearlyClaire** [comment]
If your problem is that clicking a timestamp does not open lead to the “canonical” URL of a post, that is a deliberate decision of the Mastodon interface, and the suggested flow is to click the “…” menu and select “Open original page” or “Copy link to post”.

However, I do notice that you are using a glitch-soc server. In this case, the glitch-soc front-end is available to you, and its behavior is different: clicking the timestamp will open the “canonical” URL in a new tab, which I believe is what you are asking for.  If you are not using the glitch-soc front-end and you want to use it, you can go to https://mstdn.ca/settings/flavours/glitch and click “Use this”.

**Comment by mikehamca** [comment]
You're right — I just tried it in a different browser (Opera, also built on Chromium) — deleted URL in the address bar for a given post, then clicked age of another post — and the URL did appear in the address bar. It seems to be a Vivaldi problem.

But the damn "deck/" appears in the URL even in Opera. I assume that's because I use the Advanced Web Interface.

**Comment by mikehamca** [comment]
I was calling that thing in the upper right corner the "age" instead of a "timestamp" because (in my mind) a timestamp would be constant — the time the post was added — and an age stamp would change — as the post gets older. In the Advanced Web Interface, I have an age stamp, not a timestamp. 

Ah: with the "..." menu and clicking "open original page", the page does open in a new tab and I get the canonical URL (no "deck/". That's easier than deleting "deck/" by hand.

But I could swear that in the Advanced Web Interface I did get the canonical URL for a while. I would copy the URL and paste it into a text message and would get the abbreviated post displayed, not the instance logo. I don't know why it changed, with "deck/" added to the canonical URL.

But least now I have a workaround that requires only three more steps (click age stamp, click "...", click right menu item, copy URL, close tab — instead of: click age stamp, copy URL). That's easier than deleting "deck/" by hand. Thanks.

**Comment by mikehamca** [comment]
I am the canonical user: don't read the myenu.... Apologies and thank you for your help. I am happy with it now. The change in behavior (the new "deck/" in the URL) threw me, but I see now that there's a good reason for it, and now that I have an easy way to get a good URL to paste in a message, I'm happy.

I do think the idea of replacing "No Description Added" with "Add Description" and making that a link that opens the alt-text entry box (i.e., the same link as "Edit") is good. Rather than information that something's wrong, provide the user the action step to do what's right. I would wager that "Add Description" will result in more alt text, though to know that would really require an A/B test.
